### PR TITLE
assets: expose community components

### PR DESCRIPTION
**Pull request by kpsherva**
* closes https://github.com/inveniosoftware/invenio-app-rdm/issues/2061

### CODE REVIEW

File: invenio_communities/assets/semantic-ui/js/invenio_communities/community/communitiesItems/CommunityItemMobile.js [hunk 1]
```diff
@@ -4,40 +4,30 @@
...
-          </Label>
+          <RestrictedLabel access={result.access.visibility} />
+          <CommunityTypeLabel type={communityType} />
         </Item.Extra>
         <Item.Extra>
           <Image wrapped src={result.links.logo} size="small" />
         </Item.Extra>
         <Item.Header as="h2" className="rel-mt-1">
-          <a href={`/communities/${result.id}`}>{result.metadata.title}</a>
+          <a href={`/communities/${result.slug}`}>{result.metadata.title}</a>
```
Comment (ntarocco) [inline invenio_communities/assets/semantic-ui/js/invenio_communities/community/communitiesItems/CommunityItemMobile.js:30]: 👍🏼

File: invenio_communities/assets/semantic-ui/js/invenio_communities/community/communitiesItems/CommunityItemMobile.js [hunk 2]
```diff
@@ -58,11 +48,29 @@ export const MobileCommunitiesItem = ({ result }) => {
           {i18next.t("Created: ")}
           {DateTime.fromISO(result.created).toLocaleString(i18next.language)}
         </Item.Extra>
+        {result.ui.permissions.can_update && (
+          <Item.Extra>
+            <Button
+              compact
+              size="small"
+              fluid
+              href={`/communities/${result.id}/settings`}
```
Comment (ntarocco) [inline invenio_communities/assets/semantic-ui/js/invenio_communities/community/communitiesItems/CommunityItemMobile.js:57]: Should we use `slug` here too?

File: invenio_communities/assets/semantic-ui/js/invenio_communities/community/components/CommunityTypeLabel.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import React from "react";
```
Comment (ntarocco) [inline invenio_communities/assets/semantic-ui/js/invenio_communities/community/components/CommunityTypeLabel.js:1]: WHy not moving the `access` inside `components then?

**Review by ntarocco — APPROVED** [review]

File: invenio_communities/assets/semantic-ui/js/invenio_communities/community/searchComponents/ResultsGridItemTemplate.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import React from "react";
+import { Card } from "semantic-ui-react";
+import PropTypes from "prop-types";
+
+export const ResultsGridItemTemplate = ({ result }) => {
+  return (
+    <Card fluid href={`/communities/${result.slug}`}>
+      <Card.Content>
+        <Card.Header>{result.metadata.title}</Card.Header>
+        <Card.Description>
```
Comment (jennur) [inline invenio_communities/assets/semantic-ui/js/invenio_communities/community/searchComponents/ResultsGridItemTemplate.js:10]:
```suggestion
        <Card.Description
            className="truncate-lines-2"
            dangerouslySetInnerHTML={{
              __html: result.metadata.description,
            }}
        >
```

File: invenio_communities/assets/semantic-ui/js/invenio_communities/community/communitiesItems/CommunityItemMobile.js (hunk 1 above)
Comment (jennur) [reply invenio_communities/assets/semantic-ui/js/invenio_communities/community/communitiesItems/CommunityItemMobile.js:30]: We can use `result.links.self_html` no?

Comment (kpsherva) [reply invenio_communities/assets/semantic-ui/js/invenio_communities/community/communitiesItems/CommunityItemMobile.js:30]: yes, I will add in another PR

File: invenio_communities/assets/semantic-ui/js/invenio_communities/community/communitiesItems/CommunityCompactItemComputer.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+            </Item.Header>
+
+            <Item.Description>
+              <div
+                dangerouslySetInnerHTML={{
+                  __html: _truncate(result.metadata.description, { length: 50 }),
+                }}
+              />
+            </Item.Description>
+            <Item.Extra>
```
[the remark this answers is not in the document]
Comment (kpsherva) [reply invenio_communities/assets/semantic-ui/js/invenio_communities/community/communitiesItems/CommunityCompactItemComputer.js:43]: discussed in person

File: invenio_communities/assets/semantic-ui/js/invenio_communities/community/communitiesItems/CommunityCompactItemComputer.js [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+              <RestrictedLabel access={result.access.visibility} />
+              <CommunityTypeLabel type={communityType} />
+            </Item.Extra>
+          </Item.Content>
+        </Grid.Column>
+        <Grid.Column width={4}>
+          <Item.Content>
+            <Item.Meta>
+              {result.ui.permissions.can_direct_publish && (
+                <Icon name="paper plane outline" size="big" />
```
[the remark this answers is not in the document]
Comment (kpsherva) [reply invenio_communities/assets/semantic-ui/js/invenio_communities/community/communitiesItems/CommunityCompactItemComputer.js:53]: discussed in person, the icons follow the mockups

File: invenio_communities/assets/semantic-ui/js/invenio_communities/community/communitiesItems/CommunityItemComputer.js [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+            <Item.Meta>
+              <a
+                href={result.metadata.website}
+                target="_blank"
+                rel="noopener noreferrer"
+              >
+                {result.metadata.website}
+              </a>
+            </Item.Meta>
+            <Item.Description>
```
[the remark this answers is not in the document]
Comment (kpsherva) [reply invenio_communities/assets/semantic-ui/js/invenio_communities/community/communitiesItems/CommunityItemComputer.js:42]: I checked and it was not displaying very well

**Pull request merged**